ecosystem/sf-indexer/aptos-sf-indexer/src/main.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! Indexer is used to index blockchain data into Postgres
+//!
+//! TODO: Examples
+//!
+#![forbid(unsafe_code)]
+
+use aptos_logger::{error, info};
+use aptos_sf_indexer::indexer::substream_processor::{
+    get_start_block, run_migrations, SubstreamProcessor,
+};
+use aptos_sf_indexer::proto;
+
+use anyhow::{format_err, Context, Error};
+use aptos_sf_indexer::database::new_db_pool;
+use aptos_sf_indexer::{
+    substream_processors::block_output_processor::BlockOutputSubstreamProcessor,
+    substreams::SubstreamsEndpoint,
+    substreams_stream::{BlockResponse, SubstreamsStream},
+};
+use clap::Parser;
+use futures::StreamExt;
+use prost::Message;
+use std::{env, sync::Arc};
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about, long_about = None)]
+struct IndexerArgs {
+    // URL of the firehose gRPC endpoint
+    #[clap(long)]
+    endpoint_url: String,
+
+    // Relative location of the substream wasm file (.spkg)
+    #[clap(long)]
+    package_file: String,
+
+    // Substream module name
+    #[clap(long)]
+    module_name: String,
```

Comment:
how does this work with the wasm DAG?

Reply:
Remember substreams.yaml where we define the modules? This is that module name. So if you defined say a downstream module it'll process the upstream modules in firehose (reading from cache and everything).